app/generateSMTScriptProp.js

```diff
@@ -1,22 +0,0 @@
-export default function generateSMTScriptProp(userSmt, sysSmt) {
-  const declarationsSet = new Set(); // track declared propositions
-
-  function generateDeclarations(propositions) {
-    let declarations = "";
-
-    propositions.forEach((proposition) => {
-      if (!declarationsSet.has(proposition)) {
```

Comment:
Here's the nice thing about Set's: they do not allow duplicates in them. That means, if you want to make sure there are no duplicates, just make a set, then do stuff with it.